metagraph/src/graph/alignment/aligner_seeder_methods.hpp

```diff
@@ -7,35 +7,71 @@
 
 namespace mtg {
 namespace graph {
-
-class DBGSuccinct;
-
 namespace align {
 
 class ISeeder {
   public:
-    typedef Alignment Seed;
+    typedef Matching Seed;
 
     virtual ~ISeeder() {}
 
+    virtual const DBGAlignerConfig& get_config() const = 0;
     virtual std::vector<Seed> get_seeds() const = 0;
     virtual size_t get_num_matches() const = 0;
+
+    virtual std::vector<Alignment> get_alignments() const {
+        std::vector<Alignment> alignments;
+        std::vector<Seed> seeds = get_seeds();
+        alignments.reserve(seeds.size());
+        for (Seed seed : seeds) {
+            alignments.emplace_back(std::move(seed), get_config());
+        }
```

Comment:
If I write it as `Seed&&` instead, it doesn't compile because of issues casting a rvalue to a value. I think if I just do `auto&&` that should handle all cases.